### PR TITLE
Feature plugins GET request

### PR DESCRIPTION
**Pull request by HusseinSerag**
#### Description

Please include a brief summary of the change and include the relevant issue(s).

Closes #(issue)

#### What change does this PR introduce?

Please select the relevant option(s).

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] CI/CD (updates related to the CI/CD process)
- [ ] Documentation update (changes to docs/code comments)
- [ ] Chore (miscellaneous tasks that do not fall into the above options)

#### What is the proposed approach?

Please give a short overview/explanation on the approach taken to resolve the issue(s).

#### Checklist:

- [ ] The commit message follows our adopted [guidelines](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Testing has been done for the change(s) added (for bug fixes/features)
- [ ] Relevant comments/docs have been added/updated (for bug fixes/features)

### CODE REVIEW

**Comment by hunxjunedo** [comment]
cool, will review this tmrw (hopefully)

File: src/api/routes/pluginRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      cb(new Error('Invalid file extension'));
+    }
+  },
+});
+const router = express.Router();
+
+router.get('/', getPlugins);
+router.post(
+  '/publish',
+  //checkUserSession,
```
Comment (hunxjunedo) [inline src/api/routes/pluginRoutes.ts:34]: this shouldn't make to prod

File: src/api/routes/pluginRoutes.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import checkUserSession from '../middleware/userSessionMiddleware';
+import multer from 'multer';
+import { getFileExtension } from './themeRoutes';
+
+// multer storage configuration
+const storage = multer.memoryStorage();
+
+// file upload middleware with file type filter and limits
+const upload = multer({
+  storage: storage,
```
Comment (hunxjunedo) [inline src/api/routes/pluginRoutes.ts:12]: {storage} just works as fine, looks clean, fix elsewhere too.

**Review by hunxjunedo — CHANGES_REQUESTED** [review]

**Comment by hunxjunedo** [comment]
great effort, from now onwards please to try to break down a new feature into smaller PRs and issues for easier management and review :smiley:

File: src/api/routes/pluginRoutes.ts (hunk 1 above)
Comment (HusseinSerag) [reply src/api/routes/pluginRoutes.ts:34]: The problem is testing is tricky without stopping auth, will change it of course in a couple hours.

File: src/api/controllers/pluginController.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+  try {
+    const foundPlugins =await Plugin.findAll({
+      where: whereClause,
+      offset,
+      limit,
+      raw: true
+    })
+
+    const plugins = await Promise.all(foundPlugins.map(async (plugin) => {
```
[the remark this answers is not in the document]
Comment (HusseinSerag) [reply src/api/controllers/pluginController.ts:38]: This is currently implemented “wrong”, but we moved forward to get the front end up and going and then go back and fix it properly. Will try and find a solution

File: src/api/controllers/pluginController.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    let imageURL = '';
+    if (uploadedFile) {
+      imageURL = "localhost:9000"+'/plugins-images/' + uploadedFile.name!;
+    }
+
+    const plugin = await Plugin.create({
+      name: name,
+      description: description,
+      imageURL,
+      userId: "79c302e1-033b-4567-b7b5-33470f87f88e",
```
[the remark this answers is not in the document]
Comment (HusseinSerag) [reply src/api/controllers/pluginController.ts:110]: problem with testing, will remove it now pushing to production, will add it back for dev

File: src/api/controllers/pluginController.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (HusseinSerag) [reply src/api/controllers/pluginController.ts:38]: Hey, regarding storing object in MinIO, currently our bucket is public, so i will save the perm link directly. If we change the the buckets to be private, we will change the implementation and how to save it in the DB.

File: src/api/controllers/pluginController.ts [hunk 5]
```diff
@@ -16,42 +23,15 @@ import * as crypto from 'crypto';
...
-    const foundPlugins =await Plugin.findAll({
-      where: whereClause,
-      offset,
+    const plugins = await getPluginsService({
+      searchQuery: searchQuery as string,
       limit,
-      raw: true
-    })
-
-    const plugins = await Promise.all(foundPlugins.map(async (plugin) => {
```
Comment (hunxjunedo) [inline src/api/controllers/pluginController.ts]: is this intended? Looks like you removed the whole logic for creating a presignedurl. Is this complete ?

Comment (HusseinSerag) [reply src/api/controllers/pluginController.ts]: Yes it is intended. Currently our implementation will have the buckets be public so no need to create presigned URLs, if we decide that we dont want public URLs we would have to change a few things

**Pull request merged**